clayui.com/src/components/IconSearch/index.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * © 2020 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import ClayForm, {ClayInput} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import React, {useEffect, useState} from 'react';
+
+const spritemap = '/images/icons/icons.svg';
+
+const IconSearch = source => {
+	const [searchQuery, setSearchQuery] = useState('');
+	const [filteredIconNames, setFilteredIconNames] = useState([]);
+	const list = filteredIconNames.length ? filteredIconNames : [];
+
+	source = [...source.source];
+
+	useEffect(() => {
+		const filteredIcons = source.filter(icon => {
+			const iconPrimary = icon[Object.keys(icon)[0]].toUpperCase();
+			let iconSecondary = icon[Object.keys(icon)[1]];
+
+			iconSecondary = Array.isArray(iconSecondary)
+				? iconSecondary.map(value => value.toUpperCase())
+				: iconSecondary.toUpperCase();
+
+			return (
+				iconPrimary.includes(searchQuery.toUpperCase()) ||
+				iconSecondary.includes(searchQuery.toUpperCase())
+			);
+		});
+
+		setFilteredIconNames(filteredIcons);
+	}, [searchQuery]);
+
+	return (
+		<>
+			<ClayForm.Group>
+				<label htmlFor="iconSearch">{'Search Icons'}</label>
```

Comment:
We should also add this label content to come from the parents prop, that way it is can be customized per use, such as flags. You can have `Search Icons` as the default value though in case a label is not passed in.